### ISSUE TITLE
[BUG] Failed to pull image "ghcr.io/kubeflow/spark-operator:v1beta2-1.3.3-3.1.1

### ISSUE THREAD

**Issue by vzhao12** [GoogleCloudPlatform/spark-on-k8s-operator#1991]
## Description
Unable to Start spark job in kubenetes

- [*] ✋ I have searched the open/closed issues and my issue is not listed.

## Reproduction Code [Required]

1. Set up a new kubenetes cluster. I set up one in gcloud.
2. Get kubenetes cluster config 
3. helm repo add spark-operator https://kubeflow.github.io/spark-operator
4. helm install spark-operator spark-operator/spark-operator \
--namespace default \
--set 'image.tag=v1beta2-1.3.3-3.1.1' \
--set sparkJobNamespace=default

## Expected behavior

Spin up the spark operator pod.

## Actual behavior

Pod failed because of ImagePullBackOff

Saw the following error.

`Failed to pull image "ghcr.io/kubeflow/spark-operator:v1beta2-1.3.3-3.1.1": rpc error: code = NotFound desc = failed to pull and unpack image "ghcr.io/kubeflow/spark-operator:v1beta2-1.3.3-3.1.1": failed to resolve reference "ghcr.io/kubeflow/spark-operator:v1beta2-1.3.3-3.1.1": ghcr.io/kubeflow/spark-operator:v1beta2-1.3.3-3.1.1: not found`

The errors start at 04/13/2024 1:00 AM

### Terminal Output Screenshot(s)
<img width="683" alt="Screenshot 2024-04-17 at 3 14 30 PM" src="https://github.com/kubeflow/spark-operator/assets/155690993/cb666671-36eb-4412-9634-5e7f7ccb7410">
<img width="1809" alt="Screenshot 2024-04-17 at 3 14 38 PM" src="https://github.com/kubeflow/spark-operator/assets/155690993/546e076d-d88c-4d43-955e-b5d4be0caecc">

## Environment & Versions

- Spark Operator App version:3.1.1
- Helm Chart Version: v3.12.3
- Kubernetes Version: v1.28.7-gke.1026000
- Apache Spark version:

**Comment by vzhao12** [comment]
I checked https://github.com/kubeflow/spark-operator/pkgs/container/spark-operator
It looks like we didn't publish this version at all.   

@yuchaoran2011 Can you push this version to fix the issue? Thanks

**Comment by vzhao12** [comment]
Root cause is https://github.com/kubeflow/spark-operator/pull/1937

**Comment by zevisert** [comment]
@vzhao12 Until this is addressed, you can use images from the old registry by invoking helm with an extra option
```
--set 'image.repository=ghcr.io/googlecloudplatform/spark-operator'
```

**Comment by JunseoChoJJ** [comment]
@vzhao12 I am still getting imagepullbackoff error. does anyone have idea?
helm install my-release spark-operator/spark-operator --namespace spark-operator --create-namespace --set 'image.repository=ghcr.io/googlecloudplatform/spark-operator'
I am using this command

**Comment by iva3682** [comment]
use 'image.repository=ghcr.io/kubeflow/spark-operator' and 'image.tag=v1beta2-1.4.3-3.5.0'

**Comment by vara-bonthu** [comment]
We just released a new image update with important registry fixes. Check it out:

Image tag: https://github.com/kubeflow/spark-operator/tree/v1beta2-1.4.5-3.5.0
Helm chart: https://github.com/kubeflow/spark-operator/releases/tag/spark-operator-chart-1.2.14

Please give it a try and let us know if you encounter any issues. We're working on a new KubeFlow Spark Operator release and your testing will help make it stable! Feel free to share feedback on the Kubeflow Spark operator channel.

**Comment by zevisert** [comment]
@vara-bonthu Users will still need to `--set=image.repository=...` if they are using any tag other than `v1beta2-1.4.3-3.5.0` since previous docker images have not yet been replicated to the chart's default repository (`docker.io/kubeflow/spark-operator`). 

Still only one tag exists in the default container registry: https://hub.docker.com/r/kubeflow/spark-operator/tags

**Comment by RyanZotti** [comment]
I think you meant any tag other than `v1beta2-1.4.5-3.5.0`. The 1.4.3 version isn't available but 1.4.5 is.
